src/AllUserSleep.js

```diff
@@ -1,7 +1,87 @@
+var dayjs = require("dayjs");
+var duration = require('dayjs/plugin/duration')
+dayjs.extend(duration);
+var isBetween = require('dayjs/plugin/isBetween')
+dayjs.extend(isBetween)
+
 class AllUserSleep {
   constructor(sleepData) {
     this.sleepData = sleepData;
   }
+
+  calcAvgSleepQuality(sleepData) {
+    const totalSleepQuality = sleepData.reduce((total, num) => {
+      return total + num.sleepQuality
+    }, 0)
+    const avgSleepQuality = totalSleepQuality / sleepData.length
+    return avgSleepQuality
+  }
+
+  retrieveAllUserDataByWeek(date) {
```

Comment:
Great work with this method!! This is a really great implementation of `reduce`.

I notice some similarities to the `calcSleepOverWeek` method in `UserSleep.js`. Specifically, the pattern of taking in a date & returning all data between `day1` and `day7`! This makes me wonder... Is there a way we could extract this method from both files into a `helper` method that you import into any class that needs it? It would require a bit of a refactor to both files. In my opinion, this sort of DRY thinking helps keep your apps simple (especially as you move into React!). Are there other components in your app that will eventually need to return 1 weeks worth of data?

Happy to check out this implementation together, if y'all are interested.